pkg/api/v1/types.go

```diff
@@ -2174,6 +2177,51 @@ type NodeMetric struct {
 	CPUUsedPercentage int64 `json:"cpuUsedPercentage,omitempty"`
 }
 
+// Application represents a set of applications that are to be installed for the cluster
+// swagger:model Application
+type Application struct {
+	ObjectMeta `json:",inline"`
+
+	Spec ApplicationSpec `json:"spec"`
+}
+
+// ApplicationSpec represents the specification for an application
+// swagger:model ApplicationSpec
+type ApplicationSpec struct {
+	// Namespace describe the desired state of the namespace where application will be created.
+	Namespace NamespaceSpec `json:"namespace"`
+
+	// ApplicationRef is a reference to identify which Application should be deployed
+	ApplicationRef ApplicationRef `json:"applicationRef"`
+
+	// Values describe overrides for manifest-rendering
+	Values json.RawMessage `json:"values,omitempty"`
+}
+
+type NamespaceSpec struct {
+	// Name is the namespace to deploy the Application into
+	Name string `json:"name" required:"true"`
+
+	// Create defines whether the namespace should be created if it does not exist.
+	Create bool `json:"create" required:"true"`
+
+	// Labels of the namespace
+	// More info: http://kubernetes.io/docs/user-guide/labels
```

Comment:
I would skip the second line, it is obvious.

Reply:
Makes sense, I'll remove that from labels and annotations 👍